### PR TITLE
fix: don't pass column names twice

### PR DESCRIPTION
**Pull request by judahrand**
## What does this PR address?

This can cause issues inside Pandas. I was hitting this error before the code changes here:

```
Traceback (most recent call last):
  File "<REDACTED>/BentoML/src/bentoml/_internal/server/grpc/servicer/v1/__init__.py", line 74, in Call
    input_data = await api.input.from_proto(input_proto)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<REDACTED>/BentoML/src/bentoml/_internal/io_descriptors/pandas.py", line 679, in from_proto
    dataframe = pd.DataFrame(
                ^^^^^^^^^^^^^
  File "<REDACTED>/.venv/lib/python3.11/site-packages/pandas/core/frame.py", line 709, in __init__
    mgr = dict_to_mgr(data, index, columns, dtype=dtype, copy=copy, typ=manager)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<REDACTED>/.venv/lib/python3.11/site-packages/pandas/core/internals/construction.py", line 431, in dict_to_mgr
    arrays = Series(data, index=columns, dtype=object)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<REDACTED>/.venv/lib/python3.11/site-packages/pandas/core/series.py", line 425, in __init__
    index = ensure_index(index)
            ^^^^^^^^^^^^^^^^^^^
  File "<REDACTED>/.venv/lib/python3.11/site-packages/pandas/core/indexes/base.py", line 7125, in ensure_index
    return Index(index_like, copy=copy)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<REDACTED>/.venv/lib/python3.11/site-packages/pandas/core/indexes/base.py", line 522, in __new__
    raise cls._raise_scalar_data_error(data)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<REDACTED>/.venv/lib/python3.11/site-packages/pandas/core/indexes/base.py", line 5065, in _raise_scalar_data_error
    raise TypeError(
TypeError: Index(...) must be called with a collection of some kind, ['user_id', 'item_id'] was passed
```

Fixes #(issue)

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, `pre-commit run -a` script has passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

File: src/bentoml/_internal/io_descriptors/pandas.py [hunk 1]
```diff
@@ -66,15 +66,15 @@ def get_parquet_engine() -> str:
         )
 
 
-def _openapi_types(item: str) -> str:  # pragma: no cover
+def _openapi_types(item: t.Any) -> str:  # pragma: no cover
     # convert pandas types to OpenAPI types
-    if item.startswith("int"):
+    if pd.api.types.is_integer_dtype(item):
```
[the remark this answers is not in the document]
Comment (judahrand) [reply src/bentoml/_internal/io_descriptors/pandas.py:71]: Alternatively, could we require `pandas>=0.19`? 0.19.0 was released 2nd October 2016. It feels pretty fair to not support an almost 7 year old version of Pandas.

[the remark this answers is not in the document]
Comment (judahrand) [reply src/bentoml/_internal/io_descriptors/pandas.py:71]: Whoops hadn't seen your follow-up! Pandas 1.0 is now 3 years old. Could we even just pin to that?

File: src/bentoml/_internal/io_descriptors/pandas.py [hunk 2]
```diff
@@ -676,10 +676,7 @@ def process_columns_contents(content: pb.Series) -> dict[str, t.Any]:
             with ThreadPoolExecutor(max_workers=10) as executor:
                 futures = executor.map(process_columns_contents, field.columns)
                 data.extend([i for i in list(futures)])
-            dataframe = pd.DataFrame(
-                dict(zip(field.column_names, data)),
-                columns=t.cast(t.List[str], field.column_names),
-            )
+            dataframe = pd.DataFrame(dict(zip(field.column_names, data)))
```
Comment (aarnphm) [inline src/bentoml/_internal/io_descriptors/pandas.py:679]: I believe when I first write this, the behaviour of DataFrame then ircc requires to specify the columns regardless the input type.

Comment (judahrand) [reply src/bentoml/_internal/io_descriptors/pandas.py:679]: In `pandas>=1` at least `columns` is not needed if the data passed is a `dict`: https://pandas.pydata.org/pandas-docs/version/1.0/reference/api/pandas.DataFrame.html

**Review by sauyon — APPROVED** [review]
This LGTM! Thank you!

**Pull request merged**